pkg/lib/authn/identity/loginid/normalizer.go

```diff
@@ -111,6 +114,22 @@ func (n *UsernameNormalizer) ComputeUniqueKey(normalizeLoginID string) (string,
 	return normalizeLoginID, nil
 }
 
+type PhoneNumberNormalizer struct {
+}
+
+func (n *PhoneNumberNormalizer) Normalize(loginID string) (string, error) {
+	e164, err := phone.ParseCombinedPhoneNumber(loginID)
```

Comment:
ParseCombinedPhoneNumber will check if the phone number is valid, but I think as a normalizer, we don't need to check the validity of the input. So we can simply call `phonenumbers.Parse` and `phonenumbers.Format`.

Reply:
I want to clarify the situation here.

In pkg/util/phone/phone.go,

- ParseCombinedPhoneNumber: Call phonenumbers.Parse and phonenumbers.IsValidNumber. So it accepts only valid phone numbers.
- ParseE164ToCallingCodeAndNumber: Same as ParseCombinedPhoneNumber
- Parse: Same as ParseCombinedPhoneNumber
- EnsureE164: Same as ParseCombinedPhoneNumber

So if we use any functions from pkg/util/phone, then only valid phone numbers are accepted.

Also in this PR, the normalizer used in stdattrs was switched to use the login ID normalizer. If we change the behavior of login ID normalizer, then the behavior in stdattrs will change as well.

I suggest we keep the current behavior of only accepting valid phone numbers, leaving the behavior change in https://github.com/authgear/authgear-server/issues/4015

Reply:
Actually, I think stdattrs should not use login id normalizer, because it is not necessary to be a login id?

And I think the goal of this issue is to normalize login id, but not to add additonal validations. So what I expect is not only `IsValid`, but `IsPossible` is also not checked in the normalizer, so it is not related to #4015.

Reply:
```golang
type Parser interface {
	ParseInputPhoneNumber(input string) (e164 string, err error)
	ParseCountryCallingCodeAndNationalNumber(countryCallingCode string, nationalNumber string) (e164 string, err error)
	SplitE164(input string) (countryCallingCode string, nationalNumber string, err error)
	CheckE164(s string) error
}

// What is Legal? libphonenumbers.Parse does not return err.
// Login ID uses legalParser
var LegalParser Parser
// Other places uses LegalAndValidParser.
// What is Legal and valid? libphonenumbers.Parse does not return err, and isPhoneValid returns true.
var LegalAndValidParser Parser
```